pkg/ddl/executor.go

```diff
@@ -1035,8 +1048,123 @@ func (e *executor) CreateTable(ctx sessionctx.Context, s *ast.CreateTableStmt) (
 	if s.IfNotExists {
 		onExist = OnExistIgnore
 	}
+	if err = e.CreateTableWithInfo(ctx, schema.Name, tbInfo, involvingRef, WithOnExist(onExist)); err != nil {
+		return err
+	}
+
+	if s.Select != nil {
+		sctx, err := e.sessPool.Get()
+		if err != nil {
+			return errors.Trace(err)
+		}
+		defer e.sessPool.Put(sctx)
+
+		// if isTikvStore is true, use import into instead of insert into
+		isTikvStore := e.store.Name() == kv.TiKV.Name()
+		insertSql, err := buildInsertSql(schema.Name.L, tempTableName, s, isTikvStore)
+		if err != nil {
+			return err
+		}
+
+		// insert data into temporary table
+		if _, err = sctx.GetSQLExecutor().ExecuteInternal(e.ctx, insertSql); err != nil {
+			// if insert data into temporary table failed, drop the temporary table (like rollback)
+			dropSql := buildDropSql(schema.Name.L, tempTableName)
+			if _, dropErr := sctx.GetSQLExecutor().ExecuteInternal(e.ctx, dropSql); dropErr != nil {
+				return errors.Trace(dropErr)
+			}
+			return errors.Trace(err)
+		}
+
+		// rename the temporary table to the original table
+		renameSql := buildRenameSql(schema.Name.L, tempTableName, originTableName)
```

Comment:
we need to check if the privilege are correctly handled. For example, will this patch require extra privilege that will fail the SQL? Will this patch forget to check INSERT privilege so that the haveiour is not the same as MySQL?

Reply:
nice catch.